Mateo2nd year mini project

Alpha-Martial opened the pull request:
Makes a TalonFX motor increase in speed as the right trigger of an Xbox controller is held, decrease in speed as the left trigger is held down, and stop when the B button is pressed.

Alpha-Martial committed:
Great progress!

Added the talon subsystem and commands. Made lots of progress. I believe I have all the code I need in both commands now, and I don't want to get my hopes up, but I'm pretty sure that all I have left is to configure the talon and bind buttons.

Alpha-Martial committed:
Config and Bindings

Wrote the configuration code for the TalonFX motor and bound commands to Xbox triggers. Also began work removing example command and subsystem from all code.

4682cybears committed:
Fixed stuff

Alpha-Martial committed:
Added stop command and clamped speed

Added a command, bound to the "b" button on an Xbox controller, which sets the speed of the connected TalonFX to zero when pressed.

Clamped the speed of the TalonFX between -1.0 and 1.0.

4682cybears committed:
Cleaned up stop command in TalonSubsystem

Alpha-Martial committed:
Formatting and comments

Added comments and formatted all files to fit team 4682 standards.

josieammer commented on camp/Mateo/2ndYearMiniProject/Mateo2ndYearMiniProject/src/main/java/frc/robot/Constants.java:
```diff
@@ -0,0 +1,36 @@
+    public static final int kDriverControllerPort = 0; // The controller port
+  }
+
+  public static final double RATEOFCHANGE = 0.02; // The rate of change in speed for the TalonFX
```
we use lowerCamelCase for constants names, so this should be rateOfChange

Alpha-Martial replied:
Done

josieammer commented on camp/Mateo/2ndYearMiniProject/Mateo2ndYearMiniProject/src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -0,0 +1,79 @@
+      new CommandXboxController(OperatorConstants.kDriverControllerPort); // Creates the Xbox controller and assigns it its port
+
+  // Definition of the commands
+  private final Command upCommand = new SpeedUpCommand(talon);
```
These commands are only used each in one place down below, so it's better to not create variables for them, and just create them inside the call to the button bindings below.

josieammer commented on camp/Mateo/2ndYearMiniProject/Mateo2ndYearMiniProject/src/main/java/frc/robot/subsystems/TalonSubsystem.java:
```diff
@@ -0,0 +1,127 @@
+        }
+    }
+
+    @Override
```
remove unused method.

Alpha-Martial replied:
Done

josieammer reviewed (changes requested):
Good start! Just a few things to clean up.

Alpha-Martial commented on camp/Mateo/2ndYearMiniProject/Mateo2ndYearMiniProject/src/main/java/frc/robot/subsystems/TalonSubsystem.java:
```diff
@@ -0,0 +1,127 @@
+ * the configuration for the TalonFX and all of the methods
+ * used to change speed and run the motor.
+ */
+@SuppressWarnings("unused")
```
I deleted the unused import, but I am a little concerned, as the unused import was edu.wpi.first.wpilibj.DutyCycle, which I believe should be used in the TalonSubsystem's periodic?

Alpha-Martial committed:
Resolved PR Request

Changes made:

Removed TalonStopCommand

Bound B to a new talonStop

Removed variable declarations of commands in RobotContainer, and placed the command declarations in the button bindings.

Removed an unused import

Removed example.txt

Changed RATEOFCHANGE to lowerCamelCase (rateOfChange)

Removed suppression of "unused" warnings

josieammer reviewed (approved):
Looks good!  Approved.
